python-checks/src/main/java/org/sonar/python/checks/WeakSSLProtocolCheck.java

```diff
@@ -52,9 +59,84 @@ public void initialize(Context context) {
+    private static final String TLS_SECURITY_POLICY = "tls_security_policy";
+    private static final String SENSITIVE_TLS_SECURITY_POLICY = "Policy-Min-TLS-1-0-2019-07";
+
+    @Override
+    protected void registerFqnConsumer() {
+      // Api gateway
+      checkFqn(APIGATEWAY_FQN + ".DomainName", checkDomainName(isFqn(APIGATEWAY_FQN + ".SecurityPolicy.TLS_1_0")));
+      checkFqn(APIGATEWAYV2_FQN + ".DomainName", checkDomainName(isFqn(APIGATEWAYV2_FQN + ".SecurityPolicy.TLS_1_0")));
+      checkFqn(APIGATEWAY_FQN + ".CfnDomainName", checkDomainName(isString("TLS_1_0")));
+
+      // OpenSearch & ElasticSearch
+      checkFqn(OPENSEARCH_FQN + ".Domain", checkDomain(isFqn(OPENSEARCH_FQN + ".TLSSecurityPolicy.TLS_1_0")));
+      checkFqn(ELASTICSEARCH_FQN + ".Domain", checkDomain(isFqn(ELASTICSEARCH_FQN + ".TLSSecurityPolicy.TLS_1_0")));
+      checkFqn(OPENSEARCH_FQN + ".CfnDomain", checkCfnDomain(OPENSEARCH_FQN + ".CfnDomain.DomainEndpointOptionsProperty"));
+      checkFqn(ELASTICSEARCH_FQN + ".CfnDomain", checkCfnDomain(ELASTICSEARCH_FQN + ".CfnDomain.DomainEndpointOptionsProperty"));
+    }
+
+    private static BiConsumer<SubscriptionContext, CallExpression> checkDomainName(Predicate<Expression> predicateIssue) {
+      return (ctx, callExpression) -> getArgument(ctx, callExpression, "security_policy").ifPresent(
+        argTrace -> argTrace.addIssueIf(predicateIssue, ENFORCE_MESSAGE)
+      );
+    }
+
+    private static BiConsumer<SubscriptionContext, CallExpression> checkDomain(Predicate<Expression> predicateIssue) {
+      return (ctx, callExpression) -> getArgument(ctx, callExpression, TLS_SECURITY_POLICY).ifPresentOrElse(
+        argTrace -> argTrace.addIssueIf(predicateIssue, ENFORCE_MESSAGE),
+        () -> ctx.addIssue(callExpression.callee(), OMITTING_MESSAGE)
+      );
+    }
+
+    private static BiConsumer<SubscriptionContext, CallExpression> checkCfnDomain(String domainOptionName) {
+      return (ctx, callExpression) -> getArgument(ctx, callExpression, "domain_endpoint_options").ifPresentOrElse(
+        argTrace -> argTrace.addIssueIf(isSensitiveOptionTls(ctx, domainOptionName).or(isSensitiveDictionaryTls(ctx)), ENFORCE_MESSAGE),
+        () -> ctx.addIssue(callExpression.callee(), OMITTING_MESSAGE)
+      );
+    }
+
+    private static Predicate<Expression> isSensitiveOptionTls(SubscriptionContext ctx, String domainOptionName) {
+      return expression -> {
+        if (!isFqn(domainOptionName).test(expression)) {
+          return false;
+        }
+        if (!expression.is(Tree.Kind.CALL_EXPR)) {
+          return true;
+        }
+
+        Optional<ArgumentTrace> argTrace = getArgument(ctx, (CallExpression) expression, TLS_SECURITY_POLICY);
+        if (argTrace.isEmpty()) {
+          return true;
+        }
+
+        return argTrace.filter(trace -> trace.hasExpression(isString(SENSITIVE_TLS_SECURITY_POLICY))).isPresent();
+      };
+    }
```

Comment:
This logic somehow duplicates the mechanics of `org.sonar.python.checks.cdk.DisabledESDomainEncryptionCheck#isSensitiveOptionObj`. Maybe you can build a generic utils method for this problem.